### ISSUE TITLE
Rancher server no longer accessible because of catalog URL 404

### ISSUE THREAD

**Issue by greensheep** [rancher/rancher#16048]
**Not sure if this is the correct place for this issue, please could someone point me in the right direction if it's not.**

---

**Rancher versions:**
rancher/server or rancher/rancher: 1.6.4
rancher/agent or rancher/rancher-agent:

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)**
```
Containers: 5
 Running: 5
 Paused: 0
 Stopped: 0
Images: 7
Server Version: 17.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version:  (expected: 4ab9917febca54791c5f071a9d1f404867857fcc)
runc version: N/A (expected: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe)
init version: N/A (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.4.11-23.53.amzn1.x86_64
Operating System: Amazon Linux AMI 2016.03
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.862 GiB
Name: ip-10-0-11-117
ID: GDGQ:UTAI:LN3X:WWGD:N2ND:KDWP:HBIA:2ZMJ:IOQW:V65Z:SOWY:7EWE
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
uname -r:
4.4.11-23.53.amzn1.x86_64

cat /etc/os-release:
NAME="Amazon Linux AMI"
VERSION="2016.03"
ID="amzn"
ID_LIKE="rhel fedora"
VERSION_ID="2016.03"
PRETTY_NAME="Amazon Linux AMI 2016.03"
ANSI_COLOR="0;33"
CPE_NAME="cpe:/o:amazon:linux:2016.03:ga"
HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node server, external DB

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Starting the Rancher server

**Results:**
As of this morning, I can't login to the UI (although I can see the login page) and am getting the following repeatedly (every second or so) in the logs:
```
time="2018-10-10T09:32:47Z" level=info msg="Starting Catalog Service (port 8088, refresh interval 300 seconds)"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x6c23fa]

goroutine 28 [running]:
panic(0xa336c0, 0xc42001a060)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/rancher/catalog-service/git.RemoteShaChanged(0xc4200fec30, 0x2c, 0xc4204604da, 0x6, 0xc4200fec90, 0x28, 0xc4204622a0, 0x24, 0x9bc0c0)
	/go/src/github.com/rancher/catalog-service/git/git.go:76 +0x2aa
github.com/rancher/catalog-service/manager.(*Manager).prepareGitRepoPath(0xc420182000, 0xc4204604a4, 0x6, 0xc4204604d0, 0x9, 0xc4200fec30, 0x2c, 0xc4204604da, 0x6, 0xc4200fec90, ...)
	/go/src/github.com/rancher/catalog-service/manager/local.go:86 +0x339
github.com/rancher/catalog-service/manager.(*Manager).prepareRepoPath(0xc420182000, 0xc4204604a4, 0x6, 0xc4204604d0, 0x9, 0xc4200fec30, 0x2c, 0xc4204604da, 0x6, 0xc4200fec90, ...)
	/go/src/github.com/rancher/catalog-service/manager/local.go:32 +0x99
github.com/rancher/catalog-service/manager.(*Manager).refreshCatalog(0xc420182000, 0xc4204604a4, 0x6, 0xc4204604d0, 0x9, 0xc4200fec30, 0x2c, 0xc4204604da, 0x6, 0xc4200fec90, ...)
	/go/src/github.com/rancher/catalog-service/manager/manager.go:102 +0x82
github.com/rancher/catalog-service/manager.(*Manager).refreshConfigCatalogs(0xc420182000, 0x1, 0x0, 0x0)
	/go/src/github.com/rancher/catalog-service/manager/manager.go:73 +0x2ae
github.com/rancher/catalog-service/manager.(*Manager).RefreshAll(0xc420182000, 0x526901, 0x0, 0x0)
	/go/src/github.com/rancher/catalog-service/manager/manager.go:39 +0x38
github.com/rancher/catalog-service/cmd.refresh(0xc420182000, 0x1)
	/go/src/github.com/rancher/catalog-service/cmd/root.go:177 +0x38
github.com/rancher/catalog-service/cmd.autoRefresh(0xc420182000, 0x12c)
	/go/src/github.com/rancher/catalog-service/cmd/root.go:186 +0x4f
created by github.com/rancher/catalog-service/cmd.run
	/go/src/github.com/rancher/catalog-service/cmd/root.go:145 +0x50e
```
Everything has been running smoothly for months, nothing has changed so I'm not sure where to go from here?

**Comment by greensheep** [comment]
I managed to get it working by changing the catalog URLs from this:

https://git.rancher.io/community-catalog.git
https://git.rancher.io/rancher-catalog.git

To this:

https://github.com/rancher/community-catalog.git
https://github.com/rancher/rancher-catalog.git

The `git.rancher.io` URLs are 404'ing, which I guess is a separate issue. I'll leave this open for now but the original problem is resolved.

**Comment by superseb** [comment]
The nameservers were changed for `rancher.io`, but seems to cause some stale entries in some caching nameservers. If I spin up a new AWS instance now, I get the correct results.

The URLs always respond with 404 via web, its supposed to be fetched from Git. The NPE was fixed in v1.6.5 (https://github.com/rancher/rancher/issues/9378)
